src/api/server.rkt

```diff
@@ -53,75 +82,394 @@
                     #:timeline-disabled? [timeline-disabled? #f])
   (herbie-command command test seed pcontext profile? timeline-disabled?))
 
+(define (start-job-server)
+  (set! receptionist (make-receptionist)))
+
 ;; Starts a job for a given command object|
 (define (start-job command)
   (define job-id (compute-job-id command))
-  (if (already-computed? job-id) job-id (start-work command)))
+  (place-channel-put receptionist (list 'start receptionist command job-id))
+  (when verbose
+    (eprintf "Job ~a, Qed up for program: ~a\n" job-id (test-name (herbie-command-test command))))
+  job-id)
 
 (define (is-job-finished job-id)
-  (hash-ref *job-status* job-id #f))
+  (define-values (a b) (place-channel))
+  (place-channel-put receptionist (list 'check job-id b))
+  (when verbose
+    (eprintf "Checking if job is finished: ~a.\n" job-id))
+  (place-channel-get a))
+
+; verbose logging for debugging
+(define verbose #f) ; Maybe change to log-level and use 'verbose?
+(define (log msg)
+  (when verbose
+    ;; TODO fix string interpolation
+    (eprintf "~a\n" msg)))
```

Comment:
Oh you probably don't use it because you need to allow variary args. Like this:
```
(define (log msg . args)
  (when verbose
    (apply eprintf msg args)))
```

Reply:
Thank you so much! Makes it so much cleaner.

I will look into [logging](https://docs.racket-lang.org/reference/logging.html)